### PR TITLE
Ss 3952 update metadata for ss apis

### PR DESCRIPTION
**Pull request by sheldon-b**
Updates the SelfService API metadata to work with the newest API routes. Also updates the SelfService host computation to work with micromoo-style hosts.

### CODE REVIEW

File: ss/ssc/codegen_client.go [hunk 1]
```diff
@@ -101,10 +101,6 @@ func (loc *AccountPreferenceLocator) Index(options rsapi.APIParams) ([]*AccountP
 	if filterOpt != nil {
 		params["filter[]"] = filterOpt
 	}
-	var groupOpt = options["group"]
-	if groupOpt != nil {
-		params["group"] = groupOpt
-	}
```
Comment (arangamani) [inline ss/ssc/codegen_client.go]: Is it removed?

File: ss/ssm/ssm.go [hunk 2]
```diff
@@ -19,9 +15,3 @@ func New(h string, a rsapi.Authenticator) *API {
 	ssAPI := API{API: api}
 	return &ssAPI
 }
-
-// BuildHTTPRequest wraps the underlying rsapi implementation and simply prefixes the path with
-// the service manager path.
-func (a *API) BuildHTTPRequest(verb, path, version string, params, payload rsapi.APIParams) (*http.Request, error) {
-	return a.API.BuildHTTPRequest(verb, "/manager"+path, version, params, payload)
-}
```
Comment (arangamani) [inline ss/ssm/ssm.go]: Why was this removed?

Comment (arangamani) [inline ss/ssm/ssm.go]: I assume it is not needed anymore because we have the correct path?

**Comment by arangamani** [comment]
LGTM

File: ss/ssm/ssm.go (hunk 2 above)
Comment (sheldon-b) [inline ss/ssm/ssm.go]: Yes, because now the API JSON explicitly includes "/api/<service>" whereas before the SS paths were simply, "/collections/60073/applications" etc.
